tree_test.go

```diff
@@ -333,12 +333,6 @@ func TestBuildAndVerifyProof(t *testing.T) {
 // TestBadInputs provides malicious inputs to the functions of the package,
 // trying to trigger panics or unexpected behavior.
 func TestBadInputs(t *testing.T) {
-	// Put nil into sum.
-	a := sum(sha256.New(), nil)
-	if a != nil {
-		t.Error("sum of nil should return nil")
-	}
-
```

Comment:
I'm pretty sure we need to preserve this or we risk a hardfork on the Sia blockchain.

=/